content/docs/connect/choose-connection.md

```diff
@@ -0,0 +1,163 @@
+We are working on automatic switching between HTTP and WebSocket to as needed. Check our [roadmap](/docs/introduction/roadmap) to see what's coming soon and our Friday [Changelog](/docs/changelog) for the features-of-the-week.
+</Admonition>
+
+### Step 3: Next, choose your connection type: direct or pooled
+
+You then need to decide whether to use direct connections or pooled connections using PgBouncer:
+
+- **In general, use pooled connections whenever you can**: Pooled connections can efficiently manage high numbers of concurrent connections, supporting up to 10,000 connection. This 10,000 connection limit is most useful for serverless applications and application-side connection pools that have many open connections, but infrequent and/or short transactions.
+- **Use direct (unpooled) connections if you need persistent connections**: If your application is focused mainly on tasks like migrations or administrative operations that require stable and long-lived connections, use an unpooled connection.
+
+<Admonition type="note">
+The use of connection pooling is not a magic bullet. PgBouncer has a `default_pool_size` of 64, meaning it can handle only 64 direct connections to the database at any one time. When your application connects to PgBouncer, it joins a large pool, but only 64 connections are processed at a time. Long operations can cause these connections to time out or close before tasks are completed.
+</Admonition>
+
+For more information on these choices, see:
+
+- [Neon Serverless Driver](/docs/serverless/serverless-driver)
+- [Connection pooling](/docs/connect/connection-pooling)
+
+## Common Pitfalls
+
+Here are some key points to help you navigate potential issues.
+| Issue | Description |
+|----------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Double Pooling | Avoid using client-side pooling if you're using PgBouncer on the Neon side. Just let Neon handle the pooling to prevent retaining unused connections on the client side. If you must use client-side pooling, make sure it releases connections back to the pool early enough to avoid conflicts with PgBouncer. |
+| Understanding Limits | Don't confuse `max_connections` with `default_pool_size`.<br /><br />`max_connections` is the maximum number of concurrent connections allowed by PostgreSQL and is determined by your compute size.<br /><br />`default_pool_size` is the maximum number of connections PgBouncer will pool per user/database pair, which is set to 64 by default.<br /><br />Simply increasing your compute to get more `max_connections` may not improve performance if the bottleneck is actually on your `default_pool_size`. To increase your `default_pool_size`, contact support. |
+
+## Connection types for typical applications
```

Comment:
Not sure about this section. This kind of statement seems questionable:
"direct connections for inventory updates"
I'd leave it out unless we can validate with an authoritative source.

Reply:
moved to stash file for now